src/FieldRegistry.C

```diff
@@ -27,10 +27,19 @@ Registry()
   FieldDefVector SingleStateNodalVector = {stk::topology::NODE_RANK};
   FieldDefScalar SingleStateNodalScalar = {stk::topology::NODE_RANK};
 
+  FieldDefTpetraId TpetraId = {stk::topology::NODE_RANK};
+  FieldDefGlobalId GlobalId = {stk::topology::NODE_RANK};
+  FieldDefHypreId HypreId = {stk::topology::NODE_RANK};
+
+  // clang-format off
   static const std::map<std::string, FieldDefTypes> registry = {
     {"velocity", MultiStateNodalVector},
     {"temperature", MultiStateNodalScalar},
+    {"hypre_global_id", HypreId},
+    {"tpet_global_id", TpetraId},
```

Comment:
It seems like these should be conditional on whether hypre or trilinos-solvers are enabled or not.

Reply:
Since this is the registry I don't think it matters too much. I should test that there aren't any problems though. These are just a list of all the fields that __could__ theoretically get created so as long as the types exist it should work fine.  We could if/def these, but since we do it on the actual registration calls (and we __have__ to do it there) it is not strictly necessary here.  My opinion is it would make this portion of the code messier, so I'd rather leave them out if possible.

Reply:
Yes I agree. I hadn't been thinking of this as a static set of info rather than fields actually being registered.